h2/stream.py

```diff
@@ -737,6 +753,8 @@ def upgrade(self, client_side):
         request/response of an upgraded connection. Places the stream into an
         appropriate state.
         """
+        self.config.logger.debug("Upgrading %s", self)
```

Comment:
If you want to trigger your new `__repr__` here, wouldn’t `%r` be more appropriate?

(And everywhere else you’re logging `self` in this class)